contracts/masset/liquidator/Liquidator.sol

```diff
@@ -0,0 +1,303 @@
+
+    event LiquidationCreated(address indexed bAsset);
+    event LiquidationDeleted(address indexed bAsset);
+    event LiquidationUpdated(address indexed bAsset);
+    event LiquidationTriggered(address indexed bAsset);
+    event UniswapUpdated(address indexed uniswapAddress);
+
+    address public uniswapAddress;
+
+    struct Liquidation {
+        address     bAsset;
+        address     integration;
+        address     sellToken;
+        address[]   uniswapPath;
+        uint        amount;
+        bool        paused;
+        uint        lastTriggered;
+    }
+
+    mapping(address => Liquidation) public liquidations;
+
+    /**
+     * @dev Constructor
+     * @notice To avoid variable shadowing appended `Arg` after arguments name.
+     */
+    function initialize(
+        address     _nexus,
+        address     _uniswapAddress
+    )
+        external
+        initializer
+    {
+        InitializableModule._initialize(_nexus);
+        InitializableReentrancyGuard._initialize();
+        _initialize(_uniswapAddress);
+    }
+
+
+    /**
+    * @dev Internal initialize function, to set the Uniswap address
+    * @param _uniswapAddress   Uniswap contract address
+    */
+    function _initialize(address _uniswapAddress)
+        internal
+    {
+        uniswapAddress = _uniswapAddress;
+    }
+
+    /**
+    * @dev Create a liquidation
+    * @param _bAsset The _bAsset address that this liquidation is for
+    * @param _integration The integration contract address for the _bAsset
+    * @param _sellToken The token address to be sold
+    * @param _uniswapPath The Uniswap path as an array of addresses e.g. [COMP, WETH, DAI]
+    * @param _amount The amount of tokens to be sold
+    * @param _paused Whether the liquidation is paused
+    */
+    function createLiquidation(
+        address     _bAsset,
+        address     _integration,
+        address     _sellToken,
+        address[]   calldata _uniswapPath,
+        uint        _amount,
+        bool        _paused
+    )
+        external
+        onlyGovernor
+    {
+        require(_bAsset != address(0), "_bAsset cannot be zero address");
+        require(_integration != address(0), "integration cannot be zero address");
+        // TODO add mocks for this in tests
+        //address pToken = IPlatformIntegration(_integration).bAssetToPToken(_bAsset);
+        //require(pToken != address(0), "no pToken for this bAsset");
+
+        Liquidation storage liq = liquidations[_bAsset];
+        liq.bAsset = _bAsset;
+        liq.integration = _integration;
+        liq.sellToken = _sellToken;
+        liq.uniswapPath = _uniswapPath;
+        liq.amount = _amount;
+        liq.paused = _paused;
+
+        emit LiquidationCreated(_bAsset);
+    }
+
+    /**
+     * @dev Get a liquidation
+     * @param   _bAsset Address for the underlying bAsset
+     * @return  liquidation The liquidation data
+     */
+    function readLiquidation(address _bAsset)
```

Comment:
No real need for this given that the info is on a public var and we can read off chain anyway

Reply:
Was mainly using it for testing. Can you recommend a way to read values that are set without setting an external function for tests?